### ISSUE TITLE
Alternator coredumps on start

### ISSUE THREAD

**Issue by dekimir** [scylladb/scylla#5220]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [x] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash): 8bffb800e
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): Fedora 29

On a clean build, the command line `scylla` executes as expected, but `scylla --alternator-port=8000` aborts with coredump.  Happens in both dev and debug mode.
[success.log](https://github.com/scylladb/scylla/files/3769512/success.log)
[failure.log](https://github.com/scylladb/scylla/files/3769514/failure.log)
[scylla.yaml.gz](https://github.com/scylladb/scylla/files/3769520/scylla.yaml.gz)

**Comment by dekimir** [comment]
It looks like a race condition -- many threads are initializing the same `_callbacks` object:
```
Thread 1 "scylla" hit Breakpoint 2, alternator::server::set_routes (
    this=0x529b1bc0 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server>, r=...) at alternator/server.cc:234
1: &_callbacks = (alternator::server::alternator_callbacks_map *) 0x529b1d10 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server+336>
(gdb) c
Continuing.
[Switching to Thread 0x7fffd19ff700 (LWP 74170)]

Thread 59 "reactor-30" hit Breakpoint 2, alternator::server::set_routes (
    this=0x529b1bc0 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server>, r=...) at alternator/server.cc:234
1: &_callbacks = (alternator::server::alternator_callbacks_map *) 0x529b1d10 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server+336>
(gdb) 
Continuing.
[Switching to Thread 0x7fffd9bff700 (LWP 74150)]

Thread 39 "reactor-20" hit Breakpoint 2, alternator::server::set_routes (
    this=0x529b1bc0 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server>, r=...) at alternator/server.cc:234
1: &_callbacks = (alternator::server::alternator_callbacks_map *) 0x529b1d10 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server+336>
(gdb) 
Continuing.
[Switching to Thread 0x7fffe83ff700 (LWP 74113)]

Thread 3 "reactor-2" hit Breakpoint 2, alternator::server::set_routes (
    this=0x529b1bc0 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server>, r=...) at alternator/server.cc:234
1: &_callbacks = (alternator::server::alternator_callbacks_map *) 0x529b1d10 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server+336>
(gdb) 
Continuing.
[Switching to Thread 0x7fffdc1ff700 (LWP 74143)]

Thread 33 "reactor-17" hit Breakpoint 2, alternator::server::set_routes (
    this=0x529b1bc0 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server>, r=...) at alternator/server.cc:234
1: &_callbacks = (alternator::server::alternator_callbacks_map *) 0x529b1d10 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server+336>
(gdb) 
Continuing.
[Switching to Thread 0x7fffd4dff700 (LWP 74162)]

Thread 51 "reactor-26" hit Breakpoint 2, alternator::server::set_routes (
    this=0x529b1bc0 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server>, r=...) at alternator/server.cc:234
1: &_callbacks = (alternator::server::alternator_callbacks_map *) 0x529b1d10 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server+336>
(gdb) 
Continuing.
[Switching to Thread 0x7fffe35ff700 (LWP 74125)]

Thread 15 "reactor-8" hit Breakpoint 2, alternator::server::set_routes (
    this=0x529b1bc0 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server>, r=...) at alternator/server.cc:234
1: &_callbacks = (alternator::server::alternator_callbacks_map *) 0x529b1d10 <main::{lambda()#1}::operator()() const::{lambda()#2}::operator()() const::alternator_server+336>
```

Seastar invokes this function on all shards, without any synchronization that I can see.  In `<seastar/http/httpd.hh>`:
```c++
    future<> set_routes(std::function<void(routes& r)> fun) {
        return _server_dist->invoke_on_all([fun](http_server& server) {
            fun(server._routes);
        });
    }
```

**Comment by nyh** [comment]
Removing "backport candidate" label - Alternator is not included in 3.1, and 3.2 was branched after this patch.
